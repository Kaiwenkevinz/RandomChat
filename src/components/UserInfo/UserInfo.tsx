import {View, Text, StyleSheet} from 'react-native';
import React from 'react';
import TagComponent from '../TagComponent/TagComponent';

interface UserInfoProps {
  username?: string;
  tags?: string[];
  age?: number;
  birthday?: string;
  gender?: string;
  hometown?: string;
  email?: string;
  major?: string;
  mbti?: string;
  school?: string;
  telephoneNumber?: string;
}

const UserInfo = (props: UserInfoProps) => {
  let {
    username,
    tags,
    age,
    birthday,
    gender,
    hometown,
    email,
    major,
    mbti,
    school,
    telephoneNumber,
  } = props;

  if (!tags) {
    tags = [];
  }

  const renderField = (label: string, value?: string) => {
    if (value && value !== '') {
      return (
        <View>
          <Text style={styles.label}>{label + ':'}</Text>
          <Text style={styles.value}>{value}</Text>
        </View>
      );
    }

    return null;
  };

  return (
    <View style={styles.container}>
      {renderField('Name', username)}
      {tags.length > 0 && (
        <View>
          <Text style={styles.label}>Tags:</Text>
          <View style={styles.tagContainer}>
            {tags.map((tag, idx) => (
              <TagComponent key={idx} text={tag} />
            ))}
          </View>
        </View>
      )}
      {renderField('MBTI', mbti)}
      {renderField('Gender', gender)}
      {renderField('Age', !age ? undefined : age.toString())}
      {renderField('Hometown', hometown)}
      {renderField('Email', email)}
      {renderField('Major', major)}
      {renderField('Birthday', birthday)}
      {renderField('School', school)}
      {renderField('Phone Number', telephoneNumber)}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
  },
  label: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 4,
  },
  value: {
    fontSize: 16,
    marginBottom: 16,
  },
  tagContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    rowGap: 10,
    marginBottom: 10,
  },
  button: {
    backgroundColor: '#007aff',
    padding: 10,
    borderRadius: 5,
  },
  buttonText: {
    color: '#fff',
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

export default UserInfo;
