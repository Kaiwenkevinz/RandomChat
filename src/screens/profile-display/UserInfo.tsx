import {View, Text, StyleSheet} from 'react-native';
import React from 'react';
import {IUser} from '../../types/network/types';

interface UserInfoProps {
  user: IUser;
}

const UserInfo = ({user}: UserInfoProps) => {
  const tags = user.tags?.split(';');

  return (
    <View>
      <View style={styles.container}>
        <Text style={styles.label}>Id:</Text>
        <Text style={styles.value}>{user.id}</Text>
        <Text style={styles.label}>Name:</Text>
        <Text style={styles.value}>{user.username}</Text>
        <Text style={styles.label}>Tags:</Text>
        <View style={styles.tagContainer}>
          {tags?.map((tag, idx) => (
            <Text key={idx} style={styles.tag}>
              {tag}
            </Text>
          ))}
        </View>
        <Text style={styles.label}>Gender:</Text>
        <Text style={styles.value}>{user.gender}</Text>
        <Text style={styles.label}>Age:</Text>
        <Text style={styles.value}>{user.age}</Text>
        <Text style={styles.label}>Hometown:</Text>
        <Text style={styles.value}>{user.hometown}</Text>
        <Text style={styles.label}>Email:</Text>
        <Text style={styles.value}>{user.mail}</Text>
        <Text style={styles.label}>Major:</Text>
        <Text style={styles.value}>{user.major}</Text>
        <Text style={styles.label}>Birthday:</Text>
        <Text style={styles.value}>{user.birthday}</Text>
        <Text style={styles.label}>School:</Text>
        <Text style={styles.value}>{user.school}</Text>
        <Text style={styles.label}>MBTI:</Text>
        <Text style={styles.value}>{user.mbti}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
  },
  label: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 4,
  },
  value: {
    fontSize: 16,
    marginBottom: 16,
  },
  tagContainer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    rowGap: 10,
    marginBottom: 10,
  },
  tag: {
    color: '#FFFFFF', // 白色文本
    fontSize: 14, // 字体大小
    fontWeight: 'bold', // 粗体字
    backgroundColor: '#007AFF', // 蓝色背景
    borderRadius: 4, // 圆角边框
    paddingHorizontal: 8, // 左右留白
    paddingVertical: 4, // 上下留白
    marginHorizontal: 4, // 左右留白
  },
  button: {
    backgroundColor: '#007aff',
    padding: 10,
    borderRadius: 5,
  },
  buttonText: {
    color: '#fff',
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

export default UserInfo;
