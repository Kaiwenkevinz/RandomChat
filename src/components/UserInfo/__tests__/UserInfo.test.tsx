import renderer from 'react-test-renderer';
import React from 'react';
import UserInfo from '../UserInfo';

describe('测试 UserInfo 组件的 UI', () => {
  it('should 展示所给 user 数据相对应的 user 信息', () => {
    const obj = {
      username: 'Kevin',
      tags: ['tag1', 'tag2', 'tag3'],
      age: 25,
      gender: 'male',
      hometown: 'Mars',
      email: '123@email.com',
      major: 'Nutural Science',
      mbti: 'INTJ',
      birthday: '1996-01-01',
      school: 'Mars University',
      telephoneNumber: '123456789',
    };

    const tree = renderer.create(<UserInfo {...obj} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
