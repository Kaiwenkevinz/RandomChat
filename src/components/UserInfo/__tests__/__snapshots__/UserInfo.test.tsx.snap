// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 UserInfo 组件的 UI should 展示所给 user 数据相对应的 user 信息 1`] = `
<View>
  <View
    style={
      {
        "flex": 1,
        "padding": 16,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Name:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        Kevin
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Tags:
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "marginBottom": 10,
            "rowGap": 10,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#007AFF",
                "borderRadius": 4,
                "marginHorizontal": 4,
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              },
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                {},
              ]
            }
          >
            tag1
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#007AFF",
                "borderRadius": 4,
                "marginHorizontal": 4,
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              },
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                {},
              ]
            }
          >
            tag2
          </Text>
        </View>
        <View
          style={
            [
              {
                "backgroundColor": "#007AFF",
                "borderRadius": 4,
                "marginHorizontal": 4,
                "paddingHorizontal": 8,
                "paddingVertical": 4,
              },
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                {},
              ]
            }
          >
            tag3
          </Text>
        </View>
      </View>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        MBTI:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        INTJ
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Gender:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        male
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Age:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        25
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Hometown:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        Mars
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Email:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        123@email.com
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Major:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        Nutural Science
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Birthday:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        1996-01-01
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        School:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        Mars University
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Phone Number:
      </Text>
      <Text
        style={
          {
            "fontSize": 16,
            "marginBottom": 16,
          }
        }
      >
        123456789
      </Text>
    </View>
  </View>
</View>
`;
